### PR TITLE
[Enhancement] INSERT INTO FILES supports target_max_file_size

### PR DESCRIPTION
**Pull request by letian-jiang**
## What I'm doing:

INSERT INTO FILES supports target_max_file_size

image

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/exec/pipeline/fragment_executor.cpp [hunk 1]
```diff
@@ -1045,11 +1045,18 @@ Status FragmentExecutor::_decompose_data_sink_to_operator(RuntimeState* runtime_
         RETURN_IF_ERROR(
                 Expr::create_expr_trees(runtime_state->obj_pool(), output_exprs, &output_expr_ctxs, runtime_state));
 
+        int64_t max_file_size = 1024 * 1024 * 1024L;
+        if (target_table.__isset.target_max_file_size) {
+            max_file_size = target_table.target_max_file_size;
+        }
+        if (target_table.write_single_file) {
+            max_file_size = INT_MAX;
```
Comment (dirtysalt) [inline be/src/exec/pipeline/fragment_executor.cpp:1053]: int64_max?

File: be/src/exec/pipeline/fragment_executor.cpp [hunk 2]
```diff
@@ -1045,11 +1045,18 @@ Status FragmentExecutor::_decompose_data_sink_to_operator(RuntimeState* runtime_
         RETURN_IF_ERROR(
                 Expr::create_expr_trees(runtime_state->obj_pool(), output_exprs, &output_expr_ctxs, runtime_state));
 
+        int64_t max_file_size = 1024 * 1024 * 1024L;
```
Comment (dirtysalt) [inline be/src/exec/pipeline/fragment_executor.cpp:1048]: could it be a config item? it's the max file size if it's not specified in FE.

Comment (letian-jiang) [reply be/src/exec/pipeline/fragment_executor.cpp:1048]: it is unnecessary... this PR has provided a more flexible way.

File: be/src/exec/pipeline/fragment_executor.cpp (hunk 1 above)
Comment (letian-jiang) [reply be/src/exec/pipeline/fragment_executor.cpp]: done

File: be/src/exec/pipeline/fragment_executor.cpp (hunk 2 above)
Comment (dirtysalt) [reply be/src/exec/pipeline/fragment_executor.cpp:1048]: yes, but better to define this value in a explicitly way. how to know it's 1G if max file size is not specified by FE?

**Comment by chaoyli** [comment]
#41360

**Review by dirtysalt — APPROVED** [review]

**Pull request merged**